sql/geometry.go

```diff
@@ -237,15 +240,27 @@ func (t GeometryType) Convert(v interface{}) (interface{}, error) {
 	case string:
 		return t.Convert([]byte(inner))
 	case Point:
+		if err := t.MatchSRID(inner); err != nil {
+			return nil, err
+		}
 		return Geometry{Inner: inner}, nil
 	case Linestring:
+		if err := t.MatchSRID(inner); err != nil {
+			return nil, err
+		}
 		return Geometry{Inner: inner}, nil
 	case Polygon:
+		if err := t.MatchSRID(inner); err != nil {
+			return nil, err
+		}
 		return Geometry{Inner: inner}, nil
 	case Geometry:
+		if err := t.MatchSRID(inner.Inner); err != nil {
```

Comment:
You're right! I think it's safe to call MatchSRID with Geometry object itself and recursively check SRID on its inner object. But, Geometry object does not have its own SRID member variable because it can only be either of Point, Linestring or Polygon objects, which they have SRID defined.